### ISSUE TITLE
Search for phrases

### ISSUE THREAD

**Issue by karnak-ID** [Xyrio/RSSOwlnix#75]
Search for phrases by surrounding words with quotes doesn't work.

I'm using RSSOlw version 2.7.1-beta under Windows 10 x64.

**Comment by Xyrio** [comment]
yes the version of lucene used is very old. the latest seems to support this. but also needs changes to make it work with latest.

**Comment by karnak-ID** [comment]
Thank you for the explanation.

This Lucene you mentioned seems interesting, it even has RegExp support!

**Comment by shahrk** [comment]
@Xyrio I'd like to contribute, could you help me get started? Is there any document that I can refer to begin contributing?

**Comment by Xyrio** [comment]
well there is a docs project but no idea how old that is i never bothered reading it. i just poke at the code changing it to see what stuff in the program is changed. and a lot of googling about eclipse rcp 3.x what most code is written for. but since it runs in eclipse rcp 4.x stuff can be slightly different as there is like a compatibilty layer for supporting eclipse rcp 3.x stuff in 4.x.

the eclipse rcp stuff i could do for you so you can have a faster start like 
- adding latest lucene dependency

furthermore there are the lucene specific things:
- figure out how/what is current used from lucene. like how it indexes data. the package `org.rssowl.core.internal.persist.search` is probably a good start
- next understand how the latest lucene works. since version difference is huge with many years having passed, this will probably be very different. might also be hard to find a documentation for the old version.
- redesing stuff so it works


how to setup stuff so you can work is explained here:
https://github.com/Xyrio/RSSOwlnix/wiki/How-To-Build

how to run rssowl from within eclipse so you dont have to build every time:
https://github.com/Xyrio/RSSOwlnix/wiki/How-To-Build#run-program-from-eclipse

i will make a new branch where you can then try to do this. you probably want to have everything twice. one environment where the old version is running so you can jump into old lucene sourcecode when necessary. you will have to add the src.zip once manually then it should work. and another with the new version with source and several compilation errors in rssowlnix code.
